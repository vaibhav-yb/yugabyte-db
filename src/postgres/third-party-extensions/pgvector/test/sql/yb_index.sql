CREATE TABLE items (id bigserial PRIMARY KEY, embedding vector(3));
CREATE INDEX ON items USING ybdummyann (embedding vector_l2_ops);
INSERT INTO items VALUES (1, '[1.0, 0.4, 0.3]');
INSERT INTO items VALUES (2, '[0.001, 0.432, 0.32]');
\d items

EXPLAIN (COSTS OFF) SELECT * FROM items ORDER BY embedding <-> '[1.0, 0.4, 0.3]' LIMIT 5;
SELECT * FROM items ORDER BY embedding <-> '[1.0, 0.4, 0.3]' LIMIT 5;

EXPLAIN (COSTS OFF) SELECT embedding FROM items ORDER BY embedding <-> '[1.0, 0.4, 0.3]' LIMIT 5;
SELECT embedding FROM items ORDER BY embedding <-> '[1.0, 0.4, 0.3]' LIMIT 5;

DELETE FROM items WHERE id = 1;

SELECT * FROM items ORDER BY embedding <-> '[1.0, 0.4, 0.3]' LIMIT 5;

DROP TABLE items;

-- Create a table with a vector column of size 10
CREATE TABLE items (id serial PRIMARY KEY, embedding vector(10));

CREATE INDEX ON items USING ybdummyann (embedding vector_l2_ops);

-- Insert 30 rows of sample data with embeddings having float values between 0 and 1
INSERT INTO items (embedding) VALUES
('[0.12, 0.78, 0.56, 0.34, 0.91, 0.29, 0.47, 0.68, 0.22, 0.85]'),
('[0.33, 0.14, 0.79, 0.58, 0.07, 0.42, 0.91, 0.27, 0.36, 0.84]'),
('[0.45, 0.64, 0.27, 0.19, 0.53, 0.48, 0.91, 0.85, 0.14, 0.72]'),
('[0.24, 0.85, 0.61, 0.32, 0.78, 0.45, 0.27, 0.56, 0.11, 0.90]'),
('[0.57, 0.13, 0.84, 0.46, 0.95, 0.22, 0.38, 0.73, 0.19, 0.67]'),
('[0.67, 0.24, 0.51, 0.78, 0.32, 0.81, 0.49, 0.15, 0.93, 0.56]'),
('[0.89, 0.36, 0.42, 0.71, 0.57, 0.18, 0.64, 0.28, 0.95, 0.37]'),
('[0.72, 0.58, 0.13, 0.49, 0.81, 0.64, 0.23, 0.45, 0.76, 0.12]'),
('[0.13, 0.97, 0.42, 0.81, 0.65, 0.24, 0.56, 0.38, 0.72, 0.48]'),
('[0.55, 0.72, 0.19, 0.83, 0.31, 0.48, 0.67, 0.59, 0.11, 0.90]'),
('[0.48, 0.29, 0.78, 0.52, 0.61, 0.14, 0.93, 0.37, 0.85, 0.26]'),
('[0.21, 0.43, 0.56, 0.72, 0.49, 0.64, 0.31, 0.85, 0.12, 0.78]'),
('[0.84, 0.11, 0.97, 0.46, 0.35, 0.58, 0.73, 0.92, 0.15, 0.67]'),
('[0.37, 0.52, 0.74, 0.18, 0.69, 0.43, 0.27, 0.81, 0.56, 0.12]'),
('[0.61, 0.28, 0.45, 0.72, 0.13, 0.89, 0.38, 0.57, 0.94, 0.26]'),
('[0.93, 0.14, 0.58, 0.72, 0.35, 0.49, 0.18, 0.67, 0.28, 0.82]'),
('[0.42, 0.78, 0.56, 0.34, 0.89, 0.71, 0.25, 0.63, 0.19, 0.52]'),
('[0.64, 0.19, 0.58, 0.73, 0.11, 0.46, 0.29, 0.85, 0.92, 0.35]'),
('[0.51, 0.22, 0.67, 0.34, 0.78, 0.91, 0.42, 0.56, 0.19, 0.85]'),
('[0.29, 0.67, 0.41, 0.73, 0.15, 0.52, 0.89, 0.36, 0.48, 0.94]'),
('[0.83, 0.56, 0.19, 0.41, 0.72, 0.27, 0.49, 0.93, 0.11, 0.64]'),
('[0.47, 0.22, 0.79, 0.34, 0.55, 0.18, 0.83, 0.71, 0.92, 0.13]'),
('[0.14, 0.85, 0.67, 0.41, 0.58, 0.92, 0.19, 0.73, 0.35, 0.48]'),
('[0.63, 0.24, 0.57, 0.91, 0.36, 0.42, 0.19, 0.78, 0.53, 0.85]'),
('[0.27, 0.82, 0.13, 0.69, 0.46, 0.94, 0.31, 0.58, 0.71, 0.25]'),
('[0.57, 0.38, 0.49, 0.74, 0.19, 0.91, 0.32, 0.85, 0.14, 0.63]'),
('[0.18, 0.59, 0.93, 0.27, 0.64, 0.78, 0.46, 0.52, 0.11, 0.85]'),
('[0.91, 0.22, 0.36, 0.48, 0.74, 0.67, 0.19, 0.56, 0.35, 0.81]'),
('[0.42, 0.78, 0.35, 0.67, 0.53, 0.91, 0.18, 0.72, 0.11, 0.29]'),
('[0.61, 0.57, 0.13, 0.46, 0.84, 0.92, 0.36, 0.58, 0.71, 0.25]');

EXPLAIN (COSTS OFF) SELECT * FROM items ORDER BY embedding <-> '[0.64, 0.19, 0.58, 0.73, 0.11, 0.46, 0.29, 0.85, 0.92, 0.35]';
SELECT * FROM items ORDER BY embedding <-> '[0.64, 0.19, 0.58, 0.73, 0.11, 0.46, 0.29, 0.85, 0.92, 0.35]';

/*SeqScan(items)*/ EXPLAIN (COSTS OFF) SELECT * FROM items ORDER BY embedding <-> '[0.64, 0.19, 0.58, 0.73, 0.11, 0.46, 0.29, 0.85, 0.92, 0.35]';
/*SeqScan(items)*/ SELECT * FROM items ORDER BY embedding <-> '[0.64, 0.19, 0.58, 0.73, 0.11, 0.46, 0.29, 0.85, 0.92, 0.35]';

EXPLAIN (COSTS OFF) SELECT * FROM items ORDER BY embedding <-> '[0.53, 0.38, 0.49, 0.74, 0.19, 0.91, 0.32, 0.85, 0.14, 0.63]';
SELECT * FROM items ORDER BY embedding <-> '[0.53, 0.38, 0.49, 0.74, 0.19, 0.91, 0.32, 0.85, 0.14, 0.63]';

/*SeqScan(items)*/ EXPLAIN (COSTS OFF) SELECT * FROM items ORDER BY embedding <-> '[0.53, 0.38, 0.49, 0.74, 0.19, 0.91, 0.32, 0.85, 0.14, 0.63]';
/*SeqScan(items)*/ SELECT * FROM items ORDER BY embedding <-> '[0.53, 0.38, 0.49, 0.74, 0.19, 0.91, 0.32, 0.85, 0.14, 0.63]';

-- Wrong dimensionality, shouldn't work.
SELECT * FROM items ORDER BY embedding <-> '[1.0, 0.4, 0.3]' LIMIT 5;

DROP TABLE items;
