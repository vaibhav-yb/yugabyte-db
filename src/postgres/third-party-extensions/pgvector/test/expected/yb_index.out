CREATE TABLE items (id bigserial PRIMARY KEY, embedding vector(3));
CREATE INDEX ON items USING ybdummyann (embedding vector_l2_ops);
WARNING:  ybdummyann is meant for internal-testing only and does not yield ordered results
INSERT INTO items VALUES (1, '[1.0, 0.4, 0.3]');
INSERT INTO items VALUES (2, '[0.001, 0.432, 0.32]');
\d items
                               Table "public.items"
  Column   |   Type    | Collation | Nullable |              Default
-----------+-----------+-----------+----------+-----------------------------------
 id        | bigint    |           | not null | nextval('items_id_seq'::regclass)
 embedding | vector(3) |           |          |
Indexes:
    "items_pkey" PRIMARY KEY, lsm (id HASH)
    "items_embedding_idx" ybdummyann (embedding)

EXPLAIN (COSTS OFF) SELECT * FROM items ORDER BY embedding <-> '[1.0, 0.4, 0.3]' LIMIT 5;
                       QUERY PLAN
---------------------------------------------------------
 Limit
   ->  Index Scan using items_embedding_idx on items
         Order By: (embedding <-> '[1,0.4,0.3]'::vector)
(3 rows)

SELECT * FROM items ORDER BY embedding <-> '[1.0, 0.4, 0.3]' LIMIT 5;
 id |     embedding
----+--------------------
  1 | [1,0.4,0.3]
  2 | [0.001,0.432,0.32]
(2 rows)

EXPLAIN (COSTS OFF) SELECT embedding FROM items ORDER BY embedding <-> '[1.0, 0.4, 0.3]' LIMIT 5;
                        QUERY PLAN
----------------------------------------------------------
 Limit
   ->  Index Only Scan using items_embedding_idx on items
         Order By: (embedding <-> '[1,0.4,0.3]'::vector)
(3 rows)

SELECT embedding FROM items ORDER BY embedding <-> '[1.0, 0.4, 0.3]' LIMIT 5;
     embedding
--------------------
 [1,0.4,0.3]
 [0.001,0.432,0.32]
(2 rows)

DELETE FROM items WHERE id = 1;
SELECT * FROM items ORDER BY embedding <-> '[1.0, 0.4, 0.3]' LIMIT 5;
 id |     embedding
----+--------------------
  2 | [0.001,0.432,0.32]
(1 row)

DROP TABLE items;
-- Create a table with a vector column of size 10
CREATE TABLE items (id serial PRIMARY KEY, embedding vector(10));
CREATE INDEX ON items USING ybdummyann (embedding vector_l2_ops);
WARNING:  ybdummyann is meant for internal-testing only and does not yield ordered results
-- Insert 30 rows of sample data with embeddings having float values between 0 and 1
INSERT INTO items (embedding) VALUES
('[0.12, 0.78, 0.56, 0.34, 0.91, 0.29, 0.47, 0.68, 0.22, 0.85]'),
('[0.33, 0.14, 0.79, 0.58, 0.07, 0.42, 0.91, 0.27, 0.36, 0.84]'),
('[0.45, 0.64, 0.27, 0.19, 0.53, 0.48, 0.91, 0.85, 0.14, 0.72]'),
('[0.24, 0.85, 0.61, 0.32, 0.78, 0.45, 0.27, 0.56, 0.11, 0.90]'),
('[0.57, 0.13, 0.84, 0.46, 0.95, 0.22, 0.38, 0.73, 0.19, 0.67]'),
('[0.67, 0.24, 0.51, 0.78, 0.32, 0.81, 0.49, 0.15, 0.93, 0.56]'),
('[0.89, 0.36, 0.42, 0.71, 0.57, 0.18, 0.64, 0.28, 0.95, 0.37]'),
('[0.72, 0.58, 0.13, 0.49, 0.81, 0.64, 0.23, 0.45, 0.76, 0.12]'),
('[0.13, 0.97, 0.42, 0.81, 0.65, 0.24, 0.56, 0.38, 0.72, 0.48]'),
('[0.55, 0.72, 0.19, 0.83, 0.31, 0.48, 0.67, 0.59, 0.11, 0.90]'),
('[0.48, 0.29, 0.78, 0.52, 0.61, 0.14, 0.93, 0.37, 0.85, 0.26]'),
('[0.21, 0.43, 0.56, 0.72, 0.49, 0.64, 0.31, 0.85, 0.12, 0.78]'),
('[0.84, 0.11, 0.97, 0.46, 0.35, 0.58, 0.73, 0.92, 0.15, 0.67]'),
('[0.37, 0.52, 0.74, 0.18, 0.69, 0.43, 0.27, 0.81, 0.56, 0.12]'),
('[0.61, 0.28, 0.45, 0.72, 0.13, 0.89, 0.38, 0.57, 0.94, 0.26]'),
('[0.93, 0.14, 0.58, 0.72, 0.35, 0.49, 0.18, 0.67, 0.28, 0.82]'),
('[0.42, 0.78, 0.56, 0.34, 0.89, 0.71, 0.25, 0.63, 0.19, 0.52]'),
('[0.64, 0.19, 0.58, 0.73, 0.11, 0.46, 0.29, 0.85, 0.92, 0.35]'),
('[0.51, 0.22, 0.67, 0.34, 0.78, 0.91, 0.42, 0.56, 0.19, 0.85]'),
('[0.29, 0.67, 0.41, 0.73, 0.15, 0.52, 0.89, 0.36, 0.48, 0.94]'),
('[0.83, 0.56, 0.19, 0.41, 0.72, 0.27, 0.49, 0.93, 0.11, 0.64]'),
('[0.47, 0.22, 0.79, 0.34, 0.55, 0.18, 0.83, 0.71, 0.92, 0.13]'),
('[0.14, 0.85, 0.67, 0.41, 0.58, 0.92, 0.19, 0.73, 0.35, 0.48]'),
('[0.63, 0.24, 0.57, 0.91, 0.36, 0.42, 0.19, 0.78, 0.53, 0.85]'),
('[0.27, 0.82, 0.13, 0.69, 0.46, 0.94, 0.31, 0.58, 0.71, 0.25]'),
('[0.57, 0.38, 0.49, 0.74, 0.19, 0.91, 0.32, 0.85, 0.14, 0.63]'),
('[0.18, 0.59, 0.93, 0.27, 0.64, 0.78, 0.46, 0.52, 0.11, 0.85]'),
('[0.91, 0.22, 0.36, 0.48, 0.74, 0.67, 0.19, 0.56, 0.35, 0.81]'),
('[0.42, 0.78, 0.35, 0.67, 0.53, 0.91, 0.18, 0.72, 0.11, 0.29]'),
('[0.61, 0.57, 0.13, 0.46, 0.84, 0.92, 0.36, 0.58, 0.71, 0.25]');
EXPLAIN (COSTS OFF) SELECT * FROM items ORDER BY embedding <-> '[0.64, 0.19, 0.58, 0.73, 0.11, 0.46, 0.29, 0.85, 0.92, 0.35]';
                                        QUERY PLAN                                         
-------------------------------------------------------------------------------------------
 Index Scan using items_embedding_idx on items
   Order By: (embedding <-> '[0.64,0.19,0.58,0.73,0.11,0.46,0.29,0.85,0.92,0.35]'::vector)
(2 rows)

SELECT * FROM items ORDER BY embedding <-> '[0.64, 0.19, 0.58, 0.73, 0.11, 0.46, 0.29, 0.85, 0.92, 0.35]';
 id |                      embedding                      
----+-----------------------------------------------------
 18 | [0.64,0.19,0.58,0.73,0.11,0.46,0.29,0.85,0.92,0.35]
 15 | [0.61,0.28,0.45,0.72,0.13,0.89,0.38,0.57,0.94,0.26]
 24 | [0.63,0.24,0.57,0.91,0.36,0.42,0.19,0.78,0.53,0.85]
  6 | [0.67,0.24,0.51,0.78,0.32,0.81,0.49,0.15,0.93,0.56]
 16 | [0.93,0.14,0.58,0.72,0.35,0.49,0.18,0.67,0.28,0.82]
  7 | [0.89,0.36,0.42,0.71,0.57,0.18,0.64,0.28,0.95,0.37]
 22 | [0.47,0.22,0.79,0.34,0.55,0.18,0.83,0.71,0.92,0.13]
 26 | [0.57,0.38,0.49,0.74,0.19,0.91,0.32,0.85,0.14,0.63]
 14 | [0.37,0.52,0.74,0.18,0.69,0.43,0.27,0.81,0.56,0.12]
 11 | [0.48,0.29,0.78,0.52,0.61,0.14,0.93,0.37,0.85,0.26]
  8 | [0.72,0.58,0.13,0.49,0.81,0.64,0.23,0.45,0.76,0.12]
 25 | [0.27,0.82,0.13,0.69,0.46,0.94,0.31,0.58,0.71,0.25]
 13 | [0.84,0.11,0.97,0.46,0.35,0.58,0.73,0.92,0.15,0.67]
 12 | [0.21,0.43,0.56,0.72,0.49,0.64,0.31,0.85,0.12,0.78]
 28 | [0.91,0.22,0.36,0.48,0.74,0.67,0.19,0.56,0.35,0.81]
 30 | [0.61,0.57,0.13,0.46,0.84,0.92,0.36,0.58,0.71,0.25]
  2 | [0.33,0.14,0.79,0.58,0.07,0.42,0.91,0.27,0.36,0.84]
 29 | [0.42,0.78,0.35,0.67,0.53,0.91,0.18,0.72,0.11,0.29]
 20 | [0.29,0.67,0.41,0.73,0.15,0.52,0.89,0.36,0.48,0.94]
  5 | [0.57,0.13,0.84,0.46,0.95,0.22,0.38,0.73,0.19,0.67]
  9 | [0.13,0.97,0.42,0.81,0.65,0.24,0.56,0.38,0.72,0.48]
 23 | [0.14,0.85,0.67,0.41,0.58,0.92,0.19,0.73,0.35,0.48]
 21 | [0.83,0.56,0.19,0.41,0.72,0.27,0.49,0.93,0.11,0.64]
 10 | [0.55,0.72,0.19,0.83,0.31,0.48,0.67,0.59,0.11,0.9]
 19 | [0.51,0.22,0.67,0.34,0.78,0.91,0.42,0.56,0.19,0.85]
 17 | [0.42,0.78,0.56,0.34,0.89,0.71,0.25,0.63,0.19,0.52]
  3 | [0.45,0.64,0.27,0.19,0.53,0.48,0.91,0.85,0.14,0.72]
 27 | [0.18,0.59,0.93,0.27,0.64,0.78,0.46,0.52,0.11,0.85]
  1 | [0.12,0.78,0.56,0.34,0.91,0.29,0.47,0.68,0.22,0.85]
  4 | [0.24,0.85,0.61,0.32,0.78,0.45,0.27,0.56,0.11,0.9]
(30 rows)

/*SeqScan(items)*/ EXPLAIN (COSTS OFF) SELECT * FROM items ORDER BY embedding <-> '[0.64, 0.19, 0.58, 0.73, 0.11, 0.46, 0.29, 0.85, 0.92, 0.35]';
                                        QUERY PLAN                                         
-------------------------------------------------------------------------------------------
 Index Scan using items_embedding_idx on items
   Order By: (embedding <-> '[0.64,0.19,0.58,0.73,0.11,0.46,0.29,0.85,0.92,0.35]'::vector)
(2 rows)

/*SeqScan(items)*/ SELECT * FROM items ORDER BY embedding <-> '[0.64, 0.19, 0.58, 0.73, 0.11, 0.46, 0.29, 0.85, 0.92, 0.35]';
 id |                      embedding                      
----+-----------------------------------------------------
 18 | [0.64,0.19,0.58,0.73,0.11,0.46,0.29,0.85,0.92,0.35]
 15 | [0.61,0.28,0.45,0.72,0.13,0.89,0.38,0.57,0.94,0.26]
 24 | [0.63,0.24,0.57,0.91,0.36,0.42,0.19,0.78,0.53,0.85]
  6 | [0.67,0.24,0.51,0.78,0.32,0.81,0.49,0.15,0.93,0.56]
 16 | [0.93,0.14,0.58,0.72,0.35,0.49,0.18,0.67,0.28,0.82]
  7 | [0.89,0.36,0.42,0.71,0.57,0.18,0.64,0.28,0.95,0.37]
 22 | [0.47,0.22,0.79,0.34,0.55,0.18,0.83,0.71,0.92,0.13]
 26 | [0.57,0.38,0.49,0.74,0.19,0.91,0.32,0.85,0.14,0.63]
 14 | [0.37,0.52,0.74,0.18,0.69,0.43,0.27,0.81,0.56,0.12]
 11 | [0.48,0.29,0.78,0.52,0.61,0.14,0.93,0.37,0.85,0.26]
  8 | [0.72,0.58,0.13,0.49,0.81,0.64,0.23,0.45,0.76,0.12]
 25 | [0.27,0.82,0.13,0.69,0.46,0.94,0.31,0.58,0.71,0.25]
 13 | [0.84,0.11,0.97,0.46,0.35,0.58,0.73,0.92,0.15,0.67]
 12 | [0.21,0.43,0.56,0.72,0.49,0.64,0.31,0.85,0.12,0.78]
 28 | [0.91,0.22,0.36,0.48,0.74,0.67,0.19,0.56,0.35,0.81]
 30 | [0.61,0.57,0.13,0.46,0.84,0.92,0.36,0.58,0.71,0.25]
  2 | [0.33,0.14,0.79,0.58,0.07,0.42,0.91,0.27,0.36,0.84]
 29 | [0.42,0.78,0.35,0.67,0.53,0.91,0.18,0.72,0.11,0.29]
 20 | [0.29,0.67,0.41,0.73,0.15,0.52,0.89,0.36,0.48,0.94]
  5 | [0.57,0.13,0.84,0.46,0.95,0.22,0.38,0.73,0.19,0.67]
  9 | [0.13,0.97,0.42,0.81,0.65,0.24,0.56,0.38,0.72,0.48]
 23 | [0.14,0.85,0.67,0.41,0.58,0.92,0.19,0.73,0.35,0.48]
 21 | [0.83,0.56,0.19,0.41,0.72,0.27,0.49,0.93,0.11,0.64]
 10 | [0.55,0.72,0.19,0.83,0.31,0.48,0.67,0.59,0.11,0.9]
 19 | [0.51,0.22,0.67,0.34,0.78,0.91,0.42,0.56,0.19,0.85]
 17 | [0.42,0.78,0.56,0.34,0.89,0.71,0.25,0.63,0.19,0.52]
  3 | [0.45,0.64,0.27,0.19,0.53,0.48,0.91,0.85,0.14,0.72]
 27 | [0.18,0.59,0.93,0.27,0.64,0.78,0.46,0.52,0.11,0.85]
  1 | [0.12,0.78,0.56,0.34,0.91,0.29,0.47,0.68,0.22,0.85]
  4 | [0.24,0.85,0.61,0.32,0.78,0.45,0.27,0.56,0.11,0.9]
(30 rows)

EXPLAIN (COSTS OFF) SELECT * FROM items ORDER BY embedding <-> '[0.53, 0.38, 0.49, 0.74, 0.19, 0.91, 0.32, 0.85, 0.14, 0.63]';
                                        QUERY PLAN                                         
-------------------------------------------------------------------------------------------
 Index Scan using items_embedding_idx on items
   Order By: (embedding <-> '[0.53,0.38,0.49,0.74,0.19,0.91,0.32,0.85,0.14,0.63]'::vector)
(2 rows)

SELECT * FROM items ORDER BY embedding <-> '[0.53, 0.38, 0.49, 0.74, 0.19, 0.91, 0.32, 0.85, 0.14, 0.63]';
 id |                      embedding                      
----+-----------------------------------------------------
 26 | [0.57,0.38,0.49,0.74,0.19,0.91,0.32,0.85,0.14,0.63]
 12 | [0.21,0.43,0.56,0.72,0.49,0.64,0.31,0.85,0.12,0.78]
 29 | [0.42,0.78,0.35,0.67,0.53,0.91,0.18,0.72,0.11,0.29]
 16 | [0.93,0.14,0.58,0.72,0.35,0.49,0.18,0.67,0.28,0.82]
 24 | [0.63,0.24,0.57,0.91,0.36,0.42,0.19,0.78,0.53,0.85]
 10 | [0.55,0.72,0.19,0.83,0.31,0.48,0.67,0.59,0.11,0.9]
 19 | [0.51,0.22,0.67,0.34,0.78,0.91,0.42,0.56,0.19,0.85]
 23 | [0.14,0.85,0.67,0.41,0.58,0.92,0.19,0.73,0.35,0.48]
 13 | [0.84,0.11,0.97,0.46,0.35,0.58,0.73,0.92,0.15,0.67]
 28 | [0.91,0.22,0.36,0.48,0.74,0.67,0.19,0.56,0.35,0.81]
 15 | [0.61,0.28,0.45,0.72,0.13,0.89,0.38,0.57,0.94,0.26]
 17 | [0.42,0.78,0.56,0.34,0.89,0.71,0.25,0.63,0.19,0.52]
 18 | [0.64,0.19,0.58,0.73,0.11,0.46,0.29,0.85,0.92,0.35]
 27 | [0.18,0.59,0.93,0.27,0.64,0.78,0.46,0.52,0.11,0.85]
 25 | [0.27,0.82,0.13,0.69,0.46,0.94,0.31,0.58,0.71,0.25]
 21 | [0.83,0.56,0.19,0.41,0.72,0.27,0.49,0.93,0.11,0.64]
 20 | [0.29,0.67,0.41,0.73,0.15,0.52,0.89,0.36,0.48,0.94]
  3 | [0.45,0.64,0.27,0.19,0.53,0.48,0.91,0.85,0.14,0.72]
  4 | [0.24,0.85,0.61,0.32,0.78,0.45,0.27,0.56,0.11,0.9]
  6 | [0.67,0.24,0.51,0.78,0.32,0.81,0.49,0.15,0.93,0.56]
 30 | [0.61,0.57,0.13,0.46,0.84,0.92,0.36,0.58,0.71,0.25]
  2 | [0.33,0.14,0.79,0.58,0.07,0.42,0.91,0.27,0.36,0.84]
  5 | [0.57,0.13,0.84,0.46,0.95,0.22,0.38,0.73,0.19,0.67]
 14 | [0.37,0.52,0.74,0.18,0.69,0.43,0.27,0.81,0.56,0.12]
  1 | [0.12,0.78,0.56,0.34,0.91,0.29,0.47,0.68,0.22,0.85]
  8 | [0.72,0.58,0.13,0.49,0.81,0.64,0.23,0.45,0.76,0.12]
  9 | [0.13,0.97,0.42,0.81,0.65,0.24,0.56,0.38,0.72,0.48]
  7 | [0.89,0.36,0.42,0.71,0.57,0.18,0.64,0.28,0.95,0.37]
 22 | [0.47,0.22,0.79,0.34,0.55,0.18,0.83,0.71,0.92,0.13]
 11 | [0.48,0.29,0.78,0.52,0.61,0.14,0.93,0.37,0.85,0.26]
(30 rows)

/*SeqScan(items)*/ EXPLAIN (COSTS OFF) SELECT * FROM items ORDER BY embedding <-> '[0.53, 0.38, 0.49, 0.74, 0.19, 0.91, 0.32, 0.85, 0.14, 0.63]';
                                        QUERY PLAN                                         
-------------------------------------------------------------------------------------------
 Index Scan using items_embedding_idx on items
   Order By: (embedding <-> '[0.53,0.38,0.49,0.74,0.19,0.91,0.32,0.85,0.14,0.63]'::vector)
(2 rows)

/*SeqScan(items)*/ SELECT * FROM items ORDER BY embedding <-> '[0.53, 0.38, 0.49, 0.74, 0.19, 0.91, 0.32, 0.85, 0.14, 0.63]';
 id |                      embedding                      
----+-----------------------------------------------------
 26 | [0.57,0.38,0.49,0.74,0.19,0.91,0.32,0.85,0.14,0.63]
 12 | [0.21,0.43,0.56,0.72,0.49,0.64,0.31,0.85,0.12,0.78]
 29 | [0.42,0.78,0.35,0.67,0.53,0.91,0.18,0.72,0.11,0.29]
 16 | [0.93,0.14,0.58,0.72,0.35,0.49,0.18,0.67,0.28,0.82]
 24 | [0.63,0.24,0.57,0.91,0.36,0.42,0.19,0.78,0.53,0.85]
 10 | [0.55,0.72,0.19,0.83,0.31,0.48,0.67,0.59,0.11,0.9]
 19 | [0.51,0.22,0.67,0.34,0.78,0.91,0.42,0.56,0.19,0.85]
 23 | [0.14,0.85,0.67,0.41,0.58,0.92,0.19,0.73,0.35,0.48]
 13 | [0.84,0.11,0.97,0.46,0.35,0.58,0.73,0.92,0.15,0.67]
 28 | [0.91,0.22,0.36,0.48,0.74,0.67,0.19,0.56,0.35,0.81]
 15 | [0.61,0.28,0.45,0.72,0.13,0.89,0.38,0.57,0.94,0.26]
 17 | [0.42,0.78,0.56,0.34,0.89,0.71,0.25,0.63,0.19,0.52]
 18 | [0.64,0.19,0.58,0.73,0.11,0.46,0.29,0.85,0.92,0.35]
 27 | [0.18,0.59,0.93,0.27,0.64,0.78,0.46,0.52,0.11,0.85]
 25 | [0.27,0.82,0.13,0.69,0.46,0.94,0.31,0.58,0.71,0.25]
 21 | [0.83,0.56,0.19,0.41,0.72,0.27,0.49,0.93,0.11,0.64]
 20 | [0.29,0.67,0.41,0.73,0.15,0.52,0.89,0.36,0.48,0.94]
  3 | [0.45,0.64,0.27,0.19,0.53,0.48,0.91,0.85,0.14,0.72]
  4 | [0.24,0.85,0.61,0.32,0.78,0.45,0.27,0.56,0.11,0.9]
  6 | [0.67,0.24,0.51,0.78,0.32,0.81,0.49,0.15,0.93,0.56]
 30 | [0.61,0.57,0.13,0.46,0.84,0.92,0.36,0.58,0.71,0.25]
  2 | [0.33,0.14,0.79,0.58,0.07,0.42,0.91,0.27,0.36,0.84]
  5 | [0.57,0.13,0.84,0.46,0.95,0.22,0.38,0.73,0.19,0.67]
 14 | [0.37,0.52,0.74,0.18,0.69,0.43,0.27,0.81,0.56,0.12]
  1 | [0.12,0.78,0.56,0.34,0.91,0.29,0.47,0.68,0.22,0.85]
  8 | [0.72,0.58,0.13,0.49,0.81,0.64,0.23,0.45,0.76,0.12]
  9 | [0.13,0.97,0.42,0.81,0.65,0.24,0.56,0.38,0.72,0.48]
  7 | [0.89,0.36,0.42,0.71,0.57,0.18,0.64,0.28,0.95,0.37]
 22 | [0.47,0.22,0.79,0.34,0.55,0.18,0.83,0.71,0.92,0.13]
 11 | [0.48,0.29,0.78,0.52,0.61,0.14,0.93,0.37,0.85,0.26]
(30 rows)

-- Wrong dimensionality, shouldn't work.
SELECT * FROM items ORDER BY embedding <-> '[1.0, 0.4, 0.3]' LIMIT 5;
ERROR:  different vector dimensions 10 and 3
DROP TABLE items;
